src/components/annotations/index.js

```diff
@@ -476,8 +487,15 @@ annotations.draw = function(gd, index, opt, value) {
         // make sure the arrowhead (if there is one)
         // and the annotation center are visible
         if(options.showarrow) {
-            arrowX = Lib.constrain(annPosPx.x - options.ax, 1, fullLayout.width - 1);
-            arrowY = Lib.constrain(annPosPx.y - options.ay, 1, fullLayout.height - 1);
+            if(options.axref === options.xref)
+                arrowX = Lib.constrain(annPosPx.x, 1, fullLayout.width - 1);
+            else
```

Comment:
:cow2: Oh. Please add some `{ }`.

Either as:

```js
if() {

} else {

}
```

or 

```js
if() {

}
else {

}
```

But, we allow one-liners like

```js
if() /* */
```

without braces.